orchestrator/test_runner/src/utils.rs

```diff
@@ -432,6 +433,53 @@ pub async fn print_validator_stake(contact: &Contact) {
     }
 }
 
+// Simple arguments to create a proposal with
+pub struct UpgradeProposalParams {
+    pub upgrade_height: i64,
+    pub plan_name: String,
+    pub plan_info: String,
+    pub proposal_title: String,
+    pub proposal_desc: String,
+}
+
+// Creates and submits a SoftwareUpgradeProposal to the chain, then votes yes with all validators
+pub async fn execute_upgrade_proposal(
+    contact: &Contact,
+    keys: &[ValidatorKeys],
+    timeout: Option<Duration>,
+    upgrade_params: UpgradeProposalParams,
+) {
+    let duration = match timeout {
+        Some(dur) => dur,
+        None => OPERATION_TIMEOUT,
+    };
+
+    let plan = Plan {
+        name: upgrade_params.plan_name,
+        time: None,
+        height: upgrade_params.upgrade_height,
+        info: upgrade_params.plan_info,
+    };
+    let proposal = SoftwareUpgradeProposal {
+        title: upgrade_params.proposal_title,
+        description: upgrade_params.proposal_desc,
+        plan: Some(plan),
+    };
+    let res = submit_upgrade_proposal(
+        proposal,
+        get_deposit(),
+        get_fee(None),
+        contact,
+        keys[0].validator_key,
+        Some(duration)
+    )
+    .await
+    .unwrap();
+    info!("Gov proposal executed with {:?}", res);
+
+    vote_yes_on_proposals(contact, keys, None).await;
```

Comment:
you should call wait_for_proposal_to_execute() here. That will wait until the proposal has passed for you.